test-data-generator/src/main/scala/za/co/absa/spline/testdatagen/GraphType.scala

```diff
@@ -15,14 +15,20 @@
  */
 
 package za.co.absa.spline.testdatagen
+import scala.collection.immutable
 
-object GraphType extends Enumeration {
-  type GraphType = Value
+sealed abstract class GraphType(val value: String)
```

Comment:
I would rename `value` to `name` because it's a name, and it would eliminate false relation to the `values` variable below.
```suggestion
sealed abstract class GraphType(val name: String)
```